packages/create-react-app/createReactApp.js

```diff
@@ -606,7 +606,7 @@ function isSafeToCreateProjectIn(root, name) {
 }
 
 function checkIfOnline(useYarn) {
-  if (!useYarn) {
+  if (!useYarn || process.env.https_proxy) {
```

Comment:
That seems reasonable, I'll make the change this morning.
Thinking out loud: What if the proxy is specified by IP?